src/modules/Expanders.jl

```diff
@@ -272,6 +272,102 @@ immutable DocsNodes
     nodes :: Vector{DocsNode}
 end
 
+"""
+Converts the function argument tuple `(name, type)` into a string.
+
+The tuple comes from the second return element of the `Base.arg_decl_parts(::Method)`
+and it seems they are always both `::String` (`::ASCIIString` in 0.4).
+It also appears that if the type is not declared for the method, `arg_decl_parts`
+returns an empty string.
+
+The returned string is `name::type` or just `name`, if the type is not declared.
+"""
+function _function_arg(decl)
+    n,t = decl
+    if length(t)==0
+        n
+    else
+        "$(n)::$(t)"
+    end
+end
+
+"""
+Returns a markdown paragraph (`Markdown.Paragraph`) with a list of methods.
+
+The return value will be wrapped in a `Nullable`, and a `null` is returned
+if the object in `ex` has no methods.
+"""
+function methodtable_md(mod::Module, ex, page, doc)
+    # TODO:
+    #   - fails on 0.4
+    #   - doesn't work with macros (problem with eval)
+    obj = try
+        mod.eval(ex)
+    catch e
+        warn("Unable to eval `$(ex)` in module $(mod).")
+        info("Error: "*string(e))
+        nothing
+    end
+
+    # If we were not able to eval the expression or the object doesn't have any
+    # methods (such as modules), we just return a `null`.
+    if obj === nothing || length(methods(obj)) == 0
+      return Nullable{Markdown.Paragraph}()
+    end
+
+    pkgdir = Pkg.dir()
+    s = """
+    <p><strong>Methods</strong></p>
+    <ol class="methodtable">
+    """
```

Comment:
Better to use an `IOBuffer` and `print` to it instead of using string concatenation. Then just do `Markdown.Paragraph(takebuf_string(buffer))` at the end.